src/ascii_dialog/dialog.py

```diff
@@ -0,0 +1,383 @@
+        self.filename_chooser = QComboBox()
+        self.filename_chooser.currentTextChanged.connect(self.updateCurrentFile)
+
+        self.load_button = QPushButton("Load File")
+        self.load_button.clicked.connect(self.load)
+
+        ## Dataset type selection
+        self.dataset_layout = QHBoxLayout()
+        self.dataset_label = QLabel("Dataset Type")
+        self.dataset_combobox = QComboBox()
+        for name in dataset_types:
+            self.dataset_combobox.addItem(name)
+        self.dataset_layout.addWidget(self.dataset_label)
+        self.dataset_layout.addWidget(self.dataset_combobox)
+
+        ## Seperator
+        self.sep_layout = QHBoxLayout()
+
+        self.sep_widgets: list[QWidget] = []
+        self.sep_label = QLabel('Seperators:')
+        self.sep_layout.addWidget(self.sep_label)
+        for seperator_name, value in self.seperators.items():
+            check_box = QCheckBox(seperator_name)
+            check_box.setChecked(value)
+            check_box.clicked.connect(self.seperatorToggle)
+            self.sep_widgets.append(check_box)
+            self.sep_layout.addWidget(check_box)
+
+        ## Starting Line
+        self.startline_layout = QHBoxLayout()
+        self.startline_label = QLabel('Starting Line')
+        self.startline_entry = QSpinBox()
+        self.startline_entry.valueChanged.connect(self.updateStartpos)
+        self.startline_layout.addWidget(self.startline_label)
+        self.startline_layout.addWidget(self.startline_entry)
+
+        ## Column Count
+        self.colcount_layout = QHBoxLayout()
+        self.colcount_label = QLabel('Number of Columns')
+        self.colcount_entry = QSpinBox()
+        self.colcount_entry.setMinimum(1)
+        self.colcount_entry.valueChanged.connect(self.updateColcount)
+        self.colcount_layout.addWidget(self.colcount_label)
+        self.colcount_layout.addWidget(self.colcount_entry)
+
+        ## Column Editor
+        options =  self.datasetOptions()
+        self.col_editor = ColEditor(self.colcount_entry.value(), options)
+        self.dataset_combobox.currentTextChanged.connect(self.changeDatasetType)
+        self.col_editor.column_changed.connect(self.updateColumn)
+
+        ## Data Table
+
+        self.table = QTableWidget()
+        self.table.show()
+        # Make the table readonly
+        self.table.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
+        # The table's width will always resize to fit the amount of space it has.
+        self.table.setSizeAdjustPolicy(QAbstractScrollArea.SizeAdjustPolicy.AdjustToContents)
+        # Add the context menu
+        self.table.setContextMenuPolicy(Qt.ContextMenuPolicy.CustomContextMenu)
+        self.table.customContextMenuRequested.connect(self.showContextMenu)
+
+        # Warning Label
+        self.warning_label = WarningLabel(self.requiredMissing(), self.duplicateColumns())
+
+        self.layout = QVBoxLayout(self)
```

Comment:
In SasView we try to maintain consistency in terms of GUI design and have the UI code abstracted as much as possible, with all layouts defined in the .ui files (which then get converted into .py).
Using Qt Designer is very helpful in creating well-behaving dialogs and windows and with debugging layout issues without the endless modify code -> start sasview -> test functionality -> close sasview -> modify code.... loop.

However, this is a small widget so it's probably OK to create it explicitly in Python - we have several cases like this already (DataOperationUtilityPanel, some widgets in the resolution calc. etc)

Reply:
I see your point. I have QT Designer installed but I must admit that I haven't really used it before as most UI code I've written in the past has been done explicitly in code. But I will take a look at it more closely going forward.